File: 10_12_2024_Two_small_in_every_subarray.java


class Solution {
    public int pairWithMaxSum(int[] arr) {
        int n = arr.length;
        if (n == 1) return -1;
        int ans = Integer.MIN_VALUE;

        for (int i = 1; i < n; i++) {
            ans = Math.max(ans, arr[i] + arr[i - 1]);
        }

        return ans;
    }
}

File: 10_12_2024_Two_small_in_every_subarray.py

class Solution:
    def pairWithMaxSum(self, arr):
        n = len(arr)
        if n == 1:
            return -1
        ans = float('-inf')

        for i in range(1, n):
            ans = max(ans, arr[i] + arr[i - 1])

        return ans
